changefeedccl: add metamorphic enriched envelope to all changefeed tests

asg0451 opened the pull request:
Adds a metamorphic enriched envelope to all changefeed tests, with the
ability to opt out.

Fixes #139660

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/145918)
<!-- Reviewable:end -->

asg0451 committed:
changefeedccl: add metamorphic enriched envelope to all changefeed tests

Adds a metamorphic enriched envelope to all changefeed tests, with the
ability to opt out.

Fixes #139660

Release note: None
